src/components/admin/semesters/SemesterCreateDialog.tsx

```diff
@@ -0,0 +1,187 @@
+import React from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useForm, type SubmitHandler } from "react-hook-form";
+import { z } from "zod";
+
+import { TextInput } from "@/components/TextInput";
+import { useToast } from "@/components/ui/use-toast";
+import {
+  DialogCard,
+  DialogCardFooter,
+  useDialogContext,
+} from "@/components/ui/utils/DialogUtils";
+import { compareDate, formatDateInISO, validateDate } from "@/lib/utils";
+
+//Schema
+const formSchema = z
+  .object({
+    name: z.string().min(1, "Field is required"),
+    startDate: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+    endDate: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+    breakStart: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+    breakEnd: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+  })
+  .refine((data) => compareDate(data.startDate, data.breakStart) < 0, {
+    message: "Start date must be before than break start date",
```

Comment:
check grammar here: `Start date must be before break start date`

Reply:
i suppose this also means that you should change the error messages in SemesterEditDialog? Good to have consistency. (`before` instead of `less than`)

Reply:
oops